### PR TITLE
fix(APIv2): RHINENG-11772 restrict search/sort to specific parents

### PR DESCRIPTION
**Pull request by skateman**
As we are heavily limited by scoped_search, I chose the validation feature to restrict searching to specific parent hierarchies. The solution uses the `except_parents` and `only_parents` keywords that we already have in our YAML-generated tests.

In order to make the validator access the list of current parents, they have been moved into a thread-current context variable. This is not the nicest way, but any other solution would unfortunately involve monkey-patching of `scoped_search`. The second hack is that the error message from the validation is regexp-replaced to make sure that the API response gets the correct message referring to the context and not the invalid value.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: swagger/v1/openapi.json [hunk 1]
```diff
@@ -121,33 +121,33 @@
                     "value": {
                       "data": [
                         {
-                          "id": "2c7e1f15-1ff6-4c4b-91d6-4535cc700f94",
```
Comment (romanblanco) [inline swagger/v1/openapi.json:124]: There shouldn't be changes to APIv1 spec

**Review by romanblanco — APPROVED** [review]
LGTM :+1:

File: spec/integration/v2/reports_spec.rb [hunk 2]
```diff
@@ -208,8 +208,8 @@
       operationId 'SystemReports'
       content_types
       pagination_params_v2
-      sort_params_v2(V2::Report)
-      search_params_v2(V2::Report)
+      sort_params_v2(V2::Report, except: %i[os_major_version])
+      search_params_v2(V2::Report, except: %i[os_major_version with_reported_systems])
```
Comment (LightOfHeaven1994) [inline spec/integration/v2/reports_spec.rb:212]: For some reason I'm able to filter reports under system using these filters
image

File: spec/integration/v2/systems_spec.rb [hunk 3]
```diff
@@ -149,8 +149,8 @@
       operationId 'PolicySystems'
       content_types
       pagination_params_v2
-      sort_params_v2(V2::System)
-      search_params_v2(V2::System)
+      sort_params_v2(V2::System, except: %i[os_major_version])
```
Comment (LightOfHeaven1994) [inline spec/integration/v2/systems_spec.rb:152]: Sorting is still possible:
```
!master ~/Documents/WORK/IQE/iqe-compliance-plugin> curl -XGET -u $CREDENTIALS -H 'Content-Type: application/json' --proxy "squid.corp.redhat.com:3128" "https://console.stage.redhat.com/api/compliance/v2/policies/a5a8a68d-ecb5-403d-8356-d73f76133c48/systems?sort_by=os_major_version" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3198  100  3198    0     0   1600      0  0:00:01  0:00:01 --:--:--  1600
{
  "data": [
    {
      "id": "589b3c21-52cf-4a8e-9f73-5062b9fbe69a",
      "display_name": "demo_passing_system_pci-dss_0",
      "groups": [],
      "culled_timestamp": "2024-09-24T15:24:40.030Z",
      "stale_timestamp": "2024-09-10T15:24:40.030Z",
      "stale_warning_timestamp": "2024-09-17T15:24:40.030Z",
      "updated": "2024-09-09T10:24:40.030Z",
      "insights_id": "e41a8b97-b3ad-4041-9727-d340dfb860d4",
```

**Pull request merged**